Conflicts when multiple container-image modules are present

If we have more than one container-image modules present, for example `container-image-s2i and `container-image-jib` the following error occurs:

```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:build (default) on project kubernetes-quickstart: 
Failed to build quarkus application: io.quarkus.builder.ChainBuildException: 
Multiple producers of item class io.quarkus.container.spi.ContainerImageResultBuildItem 
(io.quarkus.container.image.s2i.deployment.S2iProcessor#s2iBuildFromJar):
This is the location of the conflicting producer 
(io.quarkus.container.image.jib.deployment.JibProcessor#buildFromJar) -> [Help 1]
```

It would be nice if we could provide a better message.
Also it would be nice if we could enable, disable or explicitly select one of the two using configuration.

--- a/extensions/container-image/spi/src/main/java/io/quarkus/container/spi/ContainerImageResultBuildItem.java
+++ b/extensions/container-image/spi/src/main/java/io/quarkus/container/spi/ContainerImageResultBuildItem.java
@@ -1,8 +1,8 @@
 package io.quarkus.container.spi;
 
-import io.quarkus.builder.item.SimpleBuildItem;
+import io.quarkus.builder.item.MultiBuildItem;
 
-public final class ContainerImageResultBuildItem extends SimpleBuildItem {
+public final class ContainerImageResultBuildItem extends MultiBuildItem {
 
     private final String provider;
     private final String imageId;

--- a/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesDeployer.java
+++ b/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesDeployer.java
@@ -12,8 +12,10 @@ import java.io.FileInputStream;
 import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.nio.file.Path;
+import java.util.Arrays;
 import java.util.List;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 import javax.net.ssl.SSLHandshakeException;
 
@@ -37,21 +39,31 @@ import io.quarkus.kubernetes.spi.KubernetesDeploymentTargetBuildItem;
 public class KubernetesDeployer {
 
     private static final Logger log = Logger.getLogger(KubernetesDeployer.class);
+    private static final String[] CONTAINER_IMAGE_EXTENSIONS = { "quarkus-container-image-jib",
+            "quarkus-container-image-docker", "quarkus-container-image-s2i" };
+    private static final String CONTAINER_IMAGE_EXTENSIONS_STR = Arrays.stream(CONTAINER_IMAGE_EXTENSIONS)
+            .map(s -> "\"" + s + "\"").collect(Collectors.joining(", "));
 
     @BuildStep(onlyIf = { IsNormal.class, KubernetesDeploy.class })
     public void deploy(KubernetesClientBuildItem kubernetesClient,
-            Optional<ContainerImageResultBuildItem> containerImage,
+            List<ContainerImageResultBuildItem> containerImageResults,
             List<KubernetesDeploymentTargetBuildItem> kubernetesDeploymentTargetBuildItems,
             OutputTargetBuildItem outputTarget,
             BuildProducer<DeploymentResultBuildItem> deploymentResult) {
 
-        if (!containerImage.isPresent()) {
+        if (containerImageResults.isEmpty()) {
             throw new RuntimeException(
-                    "A Kubernetes deployment was requested but no extension was found to build a container image. Consider adding one of following extensions: \"quarkus-container-image-jib\", \"quarkus-container-image-docker\" or \"quarkus-container-image-s2i\".");
+                    "A Kubernetes deployment was requested but no extension was found to build a container image. Consider adding one of following extensions: "
+                            + CONTAINER_IMAGE_EXTENSIONS_STR + ".");
+        }
+        if (containerImageResults.size() > 1) {
+            throw new RuntimeException(
+                    "Using multiple extensions for building a container image is currently not supported. Please select one of: "
+                            + CONTAINER_IMAGE_EXTENSIONS_STR + ".");
         }
 
-        ContainerImageResultBuildItem image = containerImage.get();
-        if (!hasRegistry(image.getImageId()) && !S2I.equals(image.getProvider())) {
+        ContainerImageResultBuildItem containerImageResult = containerImageResults.get(0);
+        if (!hasRegistry(containerImageResult.getImageId()) && !S2I.equals(containerImageResult.getProvider())) {
             log.warn(
                     "A Kubernetes deployment was requested, but the container image to be built will not be pushed to any registry"
                             + " because \"quarkus.container-image.registry\" has not been set. The Kubernetes deployment will only work properly"
@@ -61,7 +73,7 @@ public class KubernetesDeployer {
         //Get any build item but if the build was s2i, use openshift
         KubernetesDeploymentTargetBuildItem deploymentTarget = kubernetesDeploymentTargetBuildItems
                 .stream()
-                .filter(d -> !"s2i".equals(containerImage.get().getProvider()) || OPENSHIFT.equals(d.getName()))
+                .filter(d -> !"s2i".equals(containerImageResult.getProvider()) || OPENSHIFT.equals(d.getName()))
                 .findFirst()
                 .orElse(new KubernetesDeploymentTargetBuildItem(KUBERNETES, DEPLOYMENT));
 

